dataretrieval/nwis.py

```diff
@@ -155,7 +163,7 @@ def _discharge_measurements(**kwargs):
     return _read_rdb(response.text), _set_metadata(response, **kwargs)
 
 
-def get_discharge_peaks(sites=None, start=None, end=None, **kwargs):
+def get_discharge_peaks(sites=None, start=None, end=None,  multi_index=None, **kwargs):
```

Comment:
You can clean up a lot of this by updating `def format_response` to accept kwargs and then pop multiindex only there instead of popping it higher up and cascading through functions that only pass it along.